### ISSUE TITLE
Add 511 Network Authentication Required status code for completeness

### ISSUE THREAD

**Issue by annp89** [kennethreitz/requests#2847]
Based on http://tools.ietf.org/html/rfc6585, looks like all the other status codes mentioned in that document are included except for 511. Any specific reason/concern?

I'd be happy to open a pull request with the following additional line if it looks reasonable to you.
```
511: ('network_authentication_required', 'network_auth', 'network_authentication'),
```

**Comment by sigmavirus24** [comment]
Please do @annp89. I think it was just an error code that we happened to miss. I don't think there was any reason or concern we had with adding it.

Cheers!

**Comment by annp89** [comment]
Awesome! Will do shortly :)

**Comment by sigmavirus24** [comment]
Closed by #2849
